src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/collator/ImpactAssessment.java

```diff
@@ -0,0 +1,154 @@
+    final Map<Dimension, Impact> impactMap = impactVector.getImpact();
+
+    impactMap.forEach((dimension, impact) -> {
+      switch (impact) {
+        case INCREASES_PRESSURE:
+          addActionToMap(perDimensionPressureIncreasingActions, actionName, dimension);
+          break;
+        case DECREASES_PRESSURE:
+          addActionToMap(perDimensionPressureDecreasingActions, actionName, dimension);
+          break;
+        case NO_IMPACT:
+          break;
+        default:
+          LOG.warn("Unknown impact value: {} encountered while adding action: {}'s impact",
+              impact, actionName);
+      }
+    });
+  }
+
+  /**
+   * Removes an action's impact from the current overall impact of all proposed actions for this
+   * node so far.
+   * @param actionName The name of the action.
+   * @param impactVector The impact vector which gives a pressure heading for various resources
+   *                     impacted by taking this action.
+   */
+  public void removeActionImpact(@NonNull final String actionName,
+      @NonNull ImpactVector impactVector) {
+    final Map<Dimension, Impact> impactMap = impactVector.getImpact();
+
+    impactMap.forEach((dimension, impact) -> {
+      switch (impact) {
+        case INCREASES_PRESSURE:
+          removeActionFromMap(perDimensionPressureIncreasingActions, actionName, dimension);
+          break;
+        case DECREASES_PRESSURE:
+          removeActionFromMap(perDimensionPressureDecreasingActions, actionName, dimension);
+          break;
+        case NO_IMPACT:
+          break;
+        default:
+          LOG.warn("Unknown impact value: {} encountered while removing action: {}'s impact",
+              impact, actionName);
+      }
+    });
+  }
+
+  /**
+   * Checks if the given impact vector aligns with the current overall impact for this node.
+   * Alignment is checked when reassessing the actions where all impacts are replayed.
+   *
+   * @param actionName The name of the action.
+   * @param impactVector The impact vector which gives a pressure heading for various resources
+   *                     impacted by taking this action.
+   * @return true if the impact vector aligns with the overall impact, false otherwise.
+   */
+  public boolean checkAlignmentAcrossDimensions(@NonNull final String actionName,
+      @NonNull final ImpactVector impactVector) {
+    boolean isAligned = true;
+
+    // If this is an action that increases pressure along some dimension for this node, and the
+    // overall assessment says there are actions that decrease pressure along those same
+    // dimensions, then this action is not aligned with the other proposed actions where the
+    // deciders are trying to reduce pressure for those dimensions.
+
+    final Map<Dimension, Impact> impactMap = impactVector.getImpact();
+    for (final Map.Entry<Dimension, Impact> entry : impactMap.entrySet()) {
+      final Impact impactOnDimension = entry.getValue();
```

Comment:
We can't use `forEach` here because we will be updating a local variable inside the body of the lambda and it needs the `isAligned` value to be effectively final. Since local variables are not effectively final, we'll need to declare an array or a list of size=1 to hold the `isAligned` value.

I chose the verbose way to avoid some confusion there.